openTSNE/affinity.py

```diff
@@ -172,8 +189,7 @@ def set_perplexity(self, new_perplexity):
         # Verify that the perplexity isn't too large
         new_perplexity = self.check_perplexity(new_perplexity)
         # Recompute the affinity matrix
-        k_neighbors = min(self.n_samples - 1, int(3 * new_perplexity))
-        if k_neighbors > self.__neighbors.shape[1]:
+        if new_perplexity > self.perplexity:
```

Comment:
I don't see the point of this change. As you implemented it now, we can choose a k that is larger than 3*perplexity. So in that case, if I instantiate `a = Aff(perplexity=30, k_neighbors=500)`, then I should be able to call `a.set_perplexity(100)` without any issues and use 300 neighbors. I should even be able to call `a.set_perplexity(500)`, since those are available.

Reply:
Fair enough, fixed such that the error only happens whenever `new_perplexity > self.__neighbors.shape[1]`.